app/lib/backend_api_logic/service_extension.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module BackendApiLogic
+  module ServiceExtension
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :backend_api_configs, inverse_of: :service, dependent: :destroy
+      has_many :backend_apis, through: :backend_api_configs, dependent: :destroy
+
+      alias_method :backend_api, :first_backend_api
+    end
+
+    def first_backend_api
+      @backend_api ||= find_or_create_first_backend_api!
+    end
+
+    def find_or_create_first_backend_api!
+      return unless account
+      config = backend_api_configs.first || create_first_backend_api_config!
+      config.backend_api
+    end
+
+    private
+
+    def create_first_backend_api_config!
+      backend_api = account.backend_apis.create!(
+        system_name: system_name,
+        name: "#{name} Backend API",
+        description: "Backend API of #{name}",
+        private_endpoint: proxy&.[]('api_backend')
+      )
+      constructor = new_record? ? :build : :create!
```

Comment:
This is very subjective and not a big deal but to me these last 3 lines would be clearer like this:
```ruby
config = backend_api_configs.build(backend_api: backend_api, path: '')
config.save! unless new_record?
```

Reply:
The difference is what `create_first_backend_api_config!` would return, so if you really want to change it, it should be
```ruby
def create_first_backend_api_config!
  backend_api = account.backend_apis.create!(
    system_name: system_name,
    name: "#{name} Backend API",
    description: "Backend API of #{name}",
    private_endpoint: proxy&.[]('api_backend')
  )
  config = backend_api_configs.build(backend_api: backend_api, path: '')
  config.save! unless new_record?
  config
end
```

Reply:
right 👍